### ISSUE TITLE
test: Minimal test suite to run against node RPC to check for breaks over the wire

### ISSUE THREAD

**Issue by renaynay** [celestiaorg/celestia-node#3959]
We need something ASAP that's minimal and covers the most important endpoints at first, and then spreads to 100% coverage of the API. We do not need the canonical API workstream to get this done, we can already do it now. 

Spin up a node instance in docker and some separate sidecar-like program that uses node's API over RPC and ensures they're all working normally.

Run a mock tendermint node as well so the cel-node has something to point at for data, we need to test writes as well.

**Comment by renaynay** [comment]
Or let's maybe look at using knuu for this as a first step.

**Comment by cmwaters** [comment]
Yeah I think this should be paired with knuu work i.e. and should ideally use raw JSON RPC queries

**Comment by Wondertan** [comment]
Well, this is what Swamp tests were for that were supposed to be run over RPC client instead of direct API calls, but iirc Ryan never had a chance to finish those.

On the raw RPC @cmwaters. There is no need for that and it will only complicate things.
We have interface and both server and client are autogenerated for it via reflection. If we make tests over the interface and ensure that those tests run over RPC client for every method, then all such breakages are gonna be detected without hussles of raw json

**Comment by Wondertan** [comment]
And this is a duplicate of https://github.com/celestiaorg/celestia-node/issues/2337 fwiw

**Comment by cmwaters** [comment]
Sure, I just want to make sure that we can guarantee that clients in other languages can use the API as expected. Raw can mean using generated interfaces if that sufficiently covers it

**Comment by renaynay** [comment]
IIRC, @chatton had a PR for this on node.
